File: java/polimorfism_java/Cilindro.java
public class Cilindro extends Figura {

    public Cilindro(String nome, float raioBase, float altura) {
        super(nome,raioBase,altura);
    }

    @Override
    public void display() {
        System.out.println("nome: "+getNome());
        System.out.println("raio: "+getDim1());
        System.out.println("altura: "+getDim2());
        System.out.println("area: "+getArea());
        System.out.println("volume: "+getVolume());
    }
    @Override
    public float getArea() {
        float areaLateral = (float)(2.0*Math.PI*getDim1()*getDim2());
        float areaBase = (float)(2.0*Math.PI*Math.pow(getDim1(),2));
        return areaLateral+areaBase;
    }
    @Override
    public float getVolume() {return (float)(Math.PI*Math.pow(getDim1(),2)*getDim2());}
}

File: exercises/polimorfism/Cilindro.cpp
#include "Cilindro.h"
#include <iostream>
#include <cmath>
using namespace std;

Cilindro::Cilindro(string nome, float base, float altura)
    : Figura(nome, base, altura) {}


void Cilindro::display(){
    cout<<"nome: "<<nome<<endl;
    cout<<"raio: "<<getDim1()<<endl;
    cout<<"altura: "<<getDim2()<<endl;
    cout<<"area: "<<Cilindro::getArea()<<endl;
    cout<<"volume: "<<Cilindro::getVolume()<<endl;
}

float Cilindro::getArea() const{
    float area_l = 2.0f*M_PI*getDim1()*getDim2();
    float area_b = 2.0f*M_PI*getDim1()*getDim1();
    return area_l+area_b;
}
float Cilindro::getVolume() const{
    return M_PI*getDim1()*getDim1()*getDim2();
}